packages/web-ui/src/sections/Settings/WorkspaceName/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+import { Workspace } from '@latitude-data/core'
+import { Input, Text } from '$ui/ds/atoms'
+import { SessionWorkspace } from '$ui/providers'
+import { useDebouncedCallback } from 'use-debounce'
+
+export default function WorkspaceName({
+  workspace,
+  update,
+}: {
+  workspace: SessionWorkspace
+  update: (payload: { name: string }) => Promise<Workspace>
+}) {
+  const onChange = async (ev: React.ChangeEvent<HTMLInputElement>) => {
+    const name = ev.target.value
+    if (!name) return
+
+    return update({ name: ev.target.value })
+  }
+  const debouncedChange = useDebouncedCallback(onChange, 500, {
+    trailing: true,
+  })
+
+  // TODO: i18n
+  return (
+    <div className='flex flex-col gap-4 max-w-[50%]'>
```

Comment:
A simple form like this is easier with normal forms + revalidatePath. You could remove a lot of the code done here

Reply:
No necessary to debounce because you don't manage input value. You let uncontrolled and on form submit action goes to the server like a plain old rails app

Reply:
let's talk offline